File: other/ARC143A.py
A = list(map(int, input().split()))
A.sort(reverse=True)

ret = 0

op = A[1] - A[2]
A[0] -= op
A[1] -= op
ret += op

if A[0] > 2 * A[1]:
    print(-1)
    exit()
op = A[0] - A[1]
A[0] -= 2 * op
A[1] -= op
A[2] -= op
ret += 2 * op

op = A[0]
ret += op

print(ret)

File: rust/src/contest/arc143/arc143_a.rs
#![allow(clippy::let_unit_value)]
#![allow(non_snake_case)]
use proconio::input;

fn main() {
    input! {
        mut A:[u64;3]
    }
    A.sort();
    A.reverse();

    if A[0] - A[1] <= A[2] {
        let mut ans = 0_u64;
        ans += A[0] - A[1];
        A[0] -= ans;
        A[2] -= ans;
        ans += A[0];
        println!("{}", ans)
    } else {
        println!("-1")
    }
}
